### ISSUE TITLE
Duplicate targets when mapping over multiple previous targets

### ISSUE THREAD

**Issue by brendanf** [ropensci/drake#713]
This bug is new since yesterday.

``` r
library(drake)
inputs <- lapply(LETTERS[1:4], as.symbol)
types <- rep(c(1, 2), each = 2)

plan <- drake_plan(
  wide1 = target(
    ez_parallel(a),
    transform = map(a = !!inputs, type = !!types) ),
  prelim = target(
    preliminary(wide1),
    transform = combine(wide1, .by = type) ),
  main = target(
    expensive_calc(prelim),
    transform = map(prelim)
    ),
  format = target(
    postformat(prelim, main),
    transform = map(prelim, main)
  ),
  trace = TRUE
)

drake_plan_source(plan)
#> drake_plan(
#>   wide1_A_1 = target(
#>     command = ez_parallel(A),
#>     a = "A",
#>     type = "1",
#>     wide1 = "wide1_A_1"
#>   ),
#>   wide1_B_1 = target(
#>     command = ez_parallel(B),
#>     a = "B",
#>     type = "1",
#>     wide1 = "wide1_B_1"
#>   ),
#>   wide1_C_2 = target(
#>     command = ez_parallel(C),
#>     a = "C",
#>     type = "2",
#>     wide1 = "wide1_C_2"
#>   ),
#>   wide1_D_2 = target(
#>     command = ez_parallel(D),
#>     a = "D",
#>     type = "2",
#>     wide1 = "wide1_D_2"
#>   ),
#>   prelim_1 = target(
#>     command = preliminary(list(wide1_A_1, wide1_B_1)),
#>     type = "1",
#>     prelim = "prelim_1"
#>   ),
#>   prelim_2 = target(
#>     command = preliminary(list(wide1_C_2, wide1_D_2)),
#>     type = "2",
#>     prelim = "prelim_2"
#>   ),
#>   main_prelim_1 = target(
#>     command = expensive_calc(prelim_1),
#>     type = "1",
#>     prelim = "prelim_1",
#>     main = "main_prelim_1"
#>   ),
#>   main_prelim_2 = target(
#>     command = expensive_calc(prelim_2),
#>     type = "2",
#>     prelim = "prelim_2",
#>     main = "main_prelim_2"
#>   ),
#>   format_prelim_1_main_prelim_1 = target(
#>     command = postformat(prelim_1, main_prelim_1),
#>     type = "1",
#>     prelim = "prelim_1",
#>     main = "main_prelim_1",
#>     format = "format_prelim_1_main_prelim_1"
#>   ),
#>   format_prelim_1_main_prelim_1.1 = target(
#>     command = postformat(prelim_1, main_prelim_1),
#>     type = "1",
#>     prelim = "prelim_1",
#>     main = "main_prelim_1",
#>     format = "format_prelim_1_main_prelim_1.1"
#>   ),
#>   format_prelim_2_main_prelim_2 = target(
#>     command = postformat(prelim_2, main_prelim_2),
#>     type = "2",
#>     prelim = "prelim_2",
#>     main = "main_prelim_2",
#>     format = "format_prelim_2_main_prelim_2"
#>   ),
#>   format_prelim_2_main_prelim_2.1 = target(
#>     command = postformat(prelim_2, main_prelim_2),
#>     type = "2",
#>     prelim = "prelim_2",
#>     main = "main_prelim_2",
#>     format = "format_prelim_2_main_prelim_2.1"
#>   )
#> )
vis_drake_graph(drake_config(plan))
```
![image](https://user-images.githubusercontent.com/16835777/52326222-2b78cf80-29e8-11e9-8d0f-8fcb1f3feee5.png)

<sup>Created on 2019-02-06 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

The targets `format_prelim_2_main_prelim_2` and `format_prelim_2_main_prelim_2.1" are identical (and the same for type 1).

**Comment by brendanf** [comment]
It looks like this was introduced in 6b1c4f8.

**Comment by wlandau** [comment]
I think I fixed it in be7adaf3b45942e9b28f0e69639efef2cca6a831. Let's keep hunting around for more edge cases.
